__tests__/app.test.js

```diff
@@ -40,4 +40,207 @@ describe('news-project', () => {
+                body: 'Great comment!'
+            }
+            return request(app)
+            .post('/api/articles/1/comments')
+            .send(newComment)
+            .expect(201)
+            .then(({body: {comment}}) => {
+                expect(comment).toEqual(expect.objectContaining({
+                    article_id: expect.any(Number),
+                    author: expect.any(String),
+                    body: expect.any(String),
+                    votes: expect.any(Number),
+                    created_at: expect.any(Number),
+                }));
+            })
+        })
+
+        test('POST: 404 - returns a message "This user does not exist" when there is no such "username" in users database', () => {
+            const newComment = {
+                username: 'ololo',
+                body: 'Great comment!'
+            }
+            return request(app)
+            .post('/api/articles/1/comments')
+            .send(newComment)
+            .expect(404)
+            .then(({body: {message}}) => {
+                expect(message).toBe('This user does not exist');
+            })
+        })
+
+        test('POST: 400 - returns a message "Bad request" when a received object is empty', () => {
+            const newComment = {}
+            return request(app)
+            .post('/api/articles/1/comments')
+            .send(newComment)
+            .expect(400)
+            .then(({body: {message}}) => {
+                expect(message).toBe('Bad Request');
+            })
+        })
+
+        test('POST: 400 - returns a message "Bad request" when a received object missing a required "body" property', () => {
+            const newComment = {
+                username: 'icellusedkars'
+            }
+            return request(app)
+            .post('/api/articles/1/comments')
+            .send(newComment)
+            .expect(400)
+            .then(({body: {message}}) => {
+                expect(message).toBe('Bad Request');
+            })
+        })
+
+        test('POST: 400 - returns a message "Bad request" when a received object missing a required "username" property', () => {
+            const newComment = {
+                body: 'Great comment!'
+            }
+            return request(app)
+            .post('/api/articles/1/comments')
+            .send(newComment)
+            .expect(400)
+            .then(({body: {message}}) => {
+                expect(message).toBe('Bad Request');
+            })
+        })
+
+        test('POST: 400 - returns a message "Bad request" when a received object has empty values', () => {
+            const newComment = {
+                username: '',
+                body: ''
+            }
+            return request(app)
+            .post('/api/articles/1/comments')
+            .send(newComment)
+            .expect(400)
+            .then(({body: {message}}) => {
+                expect(message).toBe('Bad Request');
+            })
+        })
+    })
```

Comment:
add a test that tests if the request body has additional keys and that they are ignored

Reply:
Needs a test for if the id in the url is not a number